runtime/lua/vim/_core/ex_cmd/lsp.lua

```diff
@@ -0,0 +1,223 @@
+    end
+  end
+end
+
+--- @param config_names string[]
+local function ex_lsp_enable(config_names)
+  -- Default to enabling all clients matching the filetype of the current buffer.
+  if #config_names == 0 then
+    local filetype = vim.bo.filetype
+    for _, name in ipairs(get_config_names()) do
+      local filetypes = lsp.config[name].filetypes
+      if filetypes and vim.tbl_contains(filetypes, filetype) then
+        table.insert(config_names, name)
+      end
+    end
+  end
+
+  checked_enable(config_names)
+end
+
+--- @param config_names string[]
+local function ex_lsp_disable(config_names)
+  -- Default to disabling all clients attached to the current buffer.
+  if #config_names == 0 then
+    config_names = vim
+      .iter(lsp.get_clients { bufnr = api.nvim_get_current_buf() })
+      :map(function(client)
+        return client.name
+      end)
+      :filter(function(name)
+        return lsp.config[name] ~= nil
+      end)
+      :totable()
+  end
+
+  checked_enable(config_names, false)
+end
+
+--- @param client_names string[]
+--- @return vim.lsp.Client[]
+local function get_clients_from_names(client_names)
+  -- Default to stopping all active clients attached to the current buffer.
+  if #client_names == 0 then
+    return lsp.get_clients { bufnr = api.nvim_get_current_buf() }
+  else
+    return vim
+      .iter(client_names)
+      :map(function(name)
+        local clients = lsp.get_clients { name = name }
+        if #clients == 0 then
+          vim.notify(("No active clients named '%s'"):format(name), ERROR)
+        end
+        return clients
+      end)
+      :flatten()
+      :totable()
+  end
+end
+
+--- @param client_names string[]
+local function ex_lsp_restart(client_names)
+  local clients = get_clients_from_names(client_names)
+
+  for _, client in ipairs(clients) do
+    --- @type integer[]
+    local attached_buffers = vim.tbl_keys(client.attached_buffers)
+
+    -- Reattach new client once the old one exits
+    api.nvim_create_autocmd('LspDetach', {
+      group = api.nvim_create_augroup('nvim.lsp.ex_restart_' .. client.id, {}),
+      callback = function(info)
+        if info.data.client_id ~= client.id then
+          return
+        end
+
+        local new_client_id = lsp.start(client.config, { attach = false })
+        if new_client_id then
+          for _, buffer in ipairs(attached_buffers) do
+            lsp.buf_attach_client(buffer, new_client_id)
+          end
+        end
+
+        return true -- Delete autocmd
+      end,
+    })
+
+    client:stop(client.exit_timeout)
+  end
+end
+
+--- @param client_names string[]
+local function ex_lsp_stop(client_names)
+  local clients = get_clients_from_names(client_names)
+
+  for _, client in ipairs(clients) do
+    client:stop(client.exit_timeout)
+  end
+end
+
+local actions = {
+  enable = ex_lsp_enable,
+  disable = ex_lsp_disable,
+  restart = ex_lsp_restart,
+  stop = ex_lsp_stop,
+}
+
+local available_subcmds = vim.tbl_keys(actions)
```

Comment:
For reference: builtin support for subcommands was proposed in https://github.com/neovim/neovim/pull/29116 in the Lua layer, but it probabably should be part of `nvim_create_user_command`.